net_util/src/open_tap.rs

```diff
@@ -87,11 +98,14 @@ pub fn open_tap(
                 Some(name) => Tap::open_named(name, num_rx_q, flags).map_err(Error::TapOpen)?,
                 None => Tap::new(num_rx_q).map_err(Error::TapOpen)?,
             };
-            if let Some(ip) = ip_addr {
-                tap.set_ip_addr(ip).map_err(Error::TapSetIp)?;
-            }
-            if let Some(mask) = netmask {
-                tap.set_netmask(mask).map_err(Error::TapSetNetmask)?;
+            // Don't overwrite ip configuration of existing interfaces:
+            if !tap_existed {
+                if let Some(ip) = ip_addr {
+                    tap.set_ip_addr(ip).map_err(Error::TapSetIp)?;
+                }
+                if let Some(mask) = netmask {
+                    tap.set_netmask(mask).map_err(Error::TapSetNetmask)?;
+                }
```

Comment:
Please also add a `warn!()` message to remind users that the IP configurations won't be used for existing tap devices.